File: Model/circonferenza.cpp
#include "circonferenza.h"
#include "punto.h"

Circonferenza::Circonferenza(const Punto& p, unsigned int r) : centro(p), raggio(r) {}

double Circonferenza::getArea() const {
    double a = raggio * raggio * M_PI;
    return a;

}

double Circonferenza::getPerimetro() const {
    double p = 2 * raggio * M_PI;
    return p;
}

Punto Circonferenza::getBaricentro() const {
    return centro;
}

Circonferenza* Circonferenza::clone() const {
    return new Circonferenza(*this);
}

double Circonferenza::getRaggio() const {
    return raggio;
}

File: SuperQtKalkJAVA/src/Circonferenza.java
public class Circonferenza implements Poligono {
    private Punto centro;
    private double raggio;

    public Circonferenza(Punto p, int i) {
        centro = p;
        raggio = i;
    }

    public double getArea() {
        double a = raggio * raggio * Math.PI;
        return a;
    }

    public double getPerimetro() {
        double p = 2 * raggio * Math.PI;
        return p;
    }

    public Punto getBaricentro() {
        return centro;
    }

    public double getRaggio() {
        return raggio;
    }
    
    public void setRaggio(double d) {
    	raggio = d;
    	
    }
 }
